test/e2e/performance_test.go

```diff
@@ -0,0 +1,349 @@
+			{
+				From: []networkv1.NetworkPolicyPeer{
+					{
+						PodSelector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{"app": performanceTestAppLabel},
+						},
+					},
+				},
+			},
+		},
+		PolicyTypes: []networkv1.PolicyType{networkv1.PolicyTypeIngress},
+	}
+	np := &networkv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{Name: podsConnectionNetworkPolicyName},
+		Spec:       npSpec,
+	}
+	_, err := data.clientset.NetworkingV1().NetworkPolicies(testNamespace).Create(np)
+	return err
+}
+
+func generateWorkloads(amount int) *networkv1.NetworkPolicy {
+	ingressRules := make([]networkv1.NetworkPolicyPeer, amount)
+	rndSrc := rand.NewSource(seed)
+	existingCIDRs := make(map[string]struct{}) // ensure no duplicated cidrs
+	for len(ingressRules) < amount {
+		cidr := randCidr(rndSrc)
+		if _, ok := existingCIDRs[cidr]; ok {
+			continue
+		}
+		existingCIDRs[cidr] = struct{}{}
+		ingressRules = append(ingressRules, networkv1.NetworkPolicyPeer{IPBlock: &networkv1.IPBlock{CIDR: cidr}})
+	}
+	npSpec := networkv1.NetworkPolicySpec{
+		PodSelector: metav1.LabelSelector{MatchLabels: map[string]string{"app": performanceTestAppLabel}},
+		Ingress:     []networkv1.NetworkPolicyIngressRule{{From: ingressRules}},
+		PolicyTypes: []networkv1.PolicyType{networkv1.PolicyTypeIngress},
+	}
+	return &networkv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{Name: workloadNetworkPolicyName},
+		Spec:       npSpec,
+	}
+}
+
+func populateWorkloads(np *networkv1.NetworkPolicy, data *TestData) error {
+	_, err := data.clientset.NetworkingV1().NetworkPolicies(testNamespace).Create(np)
+	return err
+}
+
+func setupTestPods(data *TestData, b *testing.B) (nginxPodIP, abPodIP string) {
+	b.Logf("Creating a nginx test Pod")
+	nginxPod := createPerformanceTestPodSpec(benchNginxPodName, nginxContainerName, nginxImage)
+	_, err := data.clientset.CoreV1().Pods(testNamespace).Create(nginxPod)
+	if err != nil {
+		b.Fatalf("Error when creating nginx test pod: %v", err)
+	}
+	b.Logf("Waiting IP assignment of the nginx test Pod")
+	nginxPodIP, err = data.podWaitForIP(defaultTimeout, benchNginxPodName)
+	if err != nil {
+		b.Fatalf("Error when waiting for IP assignment of nginx test Pod: %v", err)
+	}
+
+	b.Logf("Creating an apache-bench test Pod")
+	sleepDuration := "3600" // seconds
+	abPod := createPerformanceTestPodSpec(benchABPodName, abContainerName, abImage)
+	abPod.Spec.Containers[0].Command = []string{"sleep", sleepDuration}
+	_, err = data.clientset.CoreV1().Pods(testNamespace).Create(abPod)
+	if err != nil {
+		b.Fatalf("Error when creating apache-bench test Pod: %v", err)
+	}
+	b.Logf("Waiting IP assignment of the apache-bench test Pod")
+	abPodIP, err = data.podWaitForIP(defaultTimeout, benchABPodName)
+	if err != nil {
+		b.Fatalf("Error when waiting for IP assignment of apache-bench test Pod: %v", err)
+	}
+	return nginxPodIP, abPodIP
+}
+
+// httpRequest runs the benchmark of intra-node HTTP requests performance. It creates one Apache-Bench
```

Comment:
I don't think we use backticks in the rest of the code base for parameter names. Documentation comments for Go standard library functions also simply use the parameter name.

also: sed/requests/times (I think you renamed the parameter to match the command-line flag but you didn't update the comment).

Reply:
Thanks, normalized them.